app/controllers/teachers/groups_controller.rb

```diff
@@ -0,0 +1,52 @@
+class Teachers::GroupsController < ApplicationController
+  before_action :authenticate_user!
+ # before_action :check_role
+
+ 
+def index
+    @groups = Group.all
+  end
+ 
+  def show
+    @group = Group.find(params[:id])
+  end
+ 
+  def new
+    @group = Group.new
+  end
+ 
+  def edit
+    @group = Group.find(params[:id])
+  end
+ 
+  def create
+    @group = Group.new params.require(:group).permit(:title)
```

Comment:
follow some code conventions, for example this https://github.com/github/rubocop-github/blob/master/STYLEGUIDE.md
`Group.new params.require(:group).permit(:title)`
`Group.find(params[:id])`
use round brackets always or don't use it at all